lib/game_stats.rb

```diff
@@ -0,0 +1,58 @@
+class GameStats
+  attr_reader :games_collection
+
+  def initialize(games_collection)
+    @games_collection = games_collection
+  end
+
+  def total_score
+    @games_collection.games.map do |game|
+      game.away_goals + game.home_goals
+    end
+  end
+
+  def highest_total_score
+    total_score.max
+  end
+
+  def lowest_total_score
+    total_score.min
+  end
+
+  def percentage_home_wins
+    home_wins = @games_collection.games.count do |game|
+      game.home_goals > game.away_goals
+    end
+    (home_wins.to_f / @games_collection.games.count * 100).round(3)
+  end
+
+  def percentage_visitor_wins
+    visitor_wins = @games_collection.games.count do |game|
+      game.home_goals < game.away_goals
+    end
+    (visitor_wins.to_f / @games_collection.games.count * 100).round(3)
+  end
+
+  def percentage_ties
+    ties = @games_collection.games.count do |game|
+      game.home_goals == game.away_goals
+    end
+    (ties.to_f / @games_collection.games.count * 100).round(3)
+  end
+
+  def games_by_season
+    @games_collection.games.reduce({}) do |acc, game|
+      acc[game.season] << game if acc[game.season]
+      acc[game.season] = [game] if acc[game.season].nil?
+      acc
+    end
+  end
```

Comment:
Ww created a method to list games per season before we counted them. It reminded us of the tv-network challenge when we had to list all the shows by each actor in order to determine who prolific actors were for the network. Similarly, we create our hash with all the games per season so that we can simply count the elements within each season's array of games.